src/chart/generateCategoricalChart.tsx

```diff
@@ -1966,6 +1966,26 @@ export const generateCategoricalChart = ({
       });
     };
 
+    renderBrushY = (element: React.ReactElement) => {
+      const { margin, data } = this.props;
+      const { offset, updateId } = this.state;
+
+      // TODO: update brush when children update
+      return cloneElement(element, {
```

Comment:
Yes this is precisely the pattern we are trying to get rid of 😬 try to think of an API that does not require cloning, and replacing the props internally

Reply:
Hm on the other hand it would make sense to have the same API as the existing Brush. After all it's the same thing, isn't it.

Reply:
@PavelVanecek I couldn't find an easy answer to the problem of rendering the same component twice. Now that I am a bit more familiar with the workings of the package, it might have been easier than I thought. However, I had to make some detailed conformational changes, and played with the endIndex range. So, a separate component made sense

Reply:
I believe this does not add any extra effort for future migration, if it simply stays in line with the existing brush. We will have to migrate the brush either way.